Replace base_uint(string) with from_hex_text<> (RIPD-897)

scottschurr opened the pull request:
Removes the base_uint constructor that took a string.  Replaces
that functionality with two free functions named from_hex_text<>.
Use of from_hex_text\<\> looks like this:

auto v = from_hex_text\<uint256\>("AAA555");
static_assert (std::is_same\<decltype(v), uint256\>::value, "Huh!");

from_hex_text\<\> only operates on base_uint types.  At the moment
the list of those types include:

 o uint128,
 o uint160,
 o uint256,
 o Directory,
 o Account,
 o Currency, and
 o NodeID.

Using from_hex_text\<\> with any other types will not compile due to
an enable_if.

The important changes are in base_uint.h.  All the other changes
result from switching to use the new functions.

scottschurr commented:
Reviewers: @nbougalis (who asked for the change) and @vinniefalco (who said someone should work on it).

nbougalis commented:
Does what it claims - looks good. There's a typo in the commit message: `Use of *from_hext_text*<>`

vinniefalco commented:
:+1:

nbougalis commented:
:+1:

scottschurr commented:
I replaced the two file-scope variables with `beast::static_initializer` uses.  Assuming I fix the typo in the commit message, I think this addresses all concerns.

scottschurr commented:
Gaaah.  No good.  AppVeyor failed.  I messed up `static_initializer`.  Investigating.

scottschurr commented:
Okay.  AppVeyor and Travis are all happy with this version (removed a couple of `const`s).

scottschurr commented:
I've squashed the change that replaced the two file-scoped variables with function-scoped statics using `static_initializer`.  I also fixed the typo in the commit message.  With luck everything is ready to go.
